### DISCUSSION TITLE
Windows Users: Which installation method do you prefer?

### DISCUSSION THREAD

**Discussion by mhils** [mitmproxy/mitmproxy#5477] [General]
Hi folks,

I'm currently trying to sort out what we should recommend for installing mitmproxy on Windows. In short, we have four alternatives:

1. **Standalone Binaries.** While those intuitively sound like the best solution, the standalone binaries come with a massively increased startup time (multiple seconds) because they first need to extract some stuff to temporary directories. We'll continue to offer them, but it's really a subpar experience and thus shouldn't be our recommendation. Another downside here is that the first users who install a new version will get an "unknown binary" warning from SmartScreen unless we sign them[^1].
1. **The Windows Installer.** The installer is pretty much the same as the standalone binaries, but fast. This is because we can skip the extraction step on startup. Another benefit is that mitmproxy is added to PATH and to the start menu. The signing problem remains.
3. **Microsoft Store (new).** The third alternative is to distribute mitmproxy via the Microsoft Store. We don't advertise this publicly yet, but here's the direct link for anyone curious: https://www.microsoft.com/store/productId/9NWNDLQMNZD7 
  In short, the store has three upsides: First, this enables automatic updates[^2]. Second, the app will be signed by Microsoft, so presumably no more AV annoyances. Third, installation is most straightforward. 
  There are also three downsides to this approach: First, folks may not like the Microsoft Store for $reason. Second, there's a manual review by Microsoft for each release. While they have always gone through so far, we're relying on their review policies not to change.
6. **Linux Binaries (WSL).** This used to be a recommended alternative, but now that 1) mitmproxy console works on Windows we should discourage this. First, Networking with WSL 2 is relatively painful. Second, transparent mode / "capture all OS traffic" does not work as well with WSL setups.

I'd be curious if folks have any thoughts on this / what you think is the best experience for users. Also, how do you install mitmproxy and why do you do it that way?😃

[^1]: I'd like to not sign manually for now; it's relatively painful to implement and comes with recurring additional infrastructure cost.

[^2]: We won't include update checks in mitmproxy ourselves because that would be telemetry, and we don't want mitmproxy to phone home.

[^3]: There apparently is some form of user management if you have/pay for Azure Active Directory... just no, too much complexity.

**Comment by meitinger** [comment]
Regarding the first two options:
If we use the WiX toolset, we can create .msi files and use Burn to create bootstrappers. These can essentially act as a standalone with an one-time-extraction (and also handle non-phone-home-updates). So we could offer both options with the same toolchain.

Congrats on option 3 🎉 (Probably the preferred way from ux perspective.)

**Comment by kevin0t** [comment]
Option 2 windows installer is better.
Though last time i installed mitmproxy through "winget" package manager.
But nobody likes to install apps from microsoft store or even updates.

**Reply by mhils** [reply]
@K-techie: Do you have any particular reason why you don't like installing from the Microsoft Store?

**Reply by kevin0t** [reply]
Not any particular reason ,but it is out of habit , i never used ms store as the UI used to be sluggish and took more time to load. Also i am wary of ms store auto-updating unnecessary apps.

Whereas using an app installer package allows me to keep an offline copy of all the apps so that i can do a **clean install** after releasing i have bloated my pc with unnecessary things.

**Comment by prplecake** [comment]
Anything but the Microsoft Store. Mainly because mitmproxy never launches properly when installed through the Microsoft Store. Clicking the program in the Start Menu briefly shows a bar as if the app is updating. Attempting to launch from a pinned Taskbar icon shows an error: 
![image](https://github.com/user-attachments/assets/1e8da316-141b-417a-90cf-fe93d8a6e119)

The shortcut *should* be using my user profile's App Data as this doesn't get installed system-wide apparently.

Anyway I'm uninstalling the Microsoft Store version and just using the exe or msi as god intended.

**Reply by mhils** [reply]
@prplecake: Could you open a new issue describing what did not work for you? Thanks!
